### ISSUE TITLE
[Bug Report] v-combobox - performance - items keep re-rendering when filtered in and out

### ISSUE THREAD

**Issue by olivierkrener** [vuetifyjs/vuetify#11635]
### Environment
**Vuetify Version:** 2.3.0
**Vue Version:** 2.6.11
**Browsers:** Chrome 83.0.4103.97
**OS:** Windows 10

### Steps to reproduce
Just type in some text in the combobox and see the number of mounted items increase.

### Expected Behavior
I would expect each item to be rendered only once, and then simply hidden or shown depending on what is typed - but they are rendered again and again.

### Actual Behavior
See above

### Reproduction Link
<a href="https://codepen.io/olivierkrener/pen/MWKebOv" target="_blank">https://codepen.io/olivierkrener/pen/MWKebOv</a>

### Other comments
In my app the rendering is expensive....

**Comment by johnleider** [comment]
I'm honestly not sure what you're doing or trying to do, but the example doesn't work.

You've submitted an issue that does not contain a **valid** or **simple** reproduction. In order to effectively disposition this, we need to verify that the problem exists **outside** of your environment concisely as possible. This issue will be tagged as `needs reproduction` for the next 48 hours before being closed.

You can find information on how to contribute to Vuetify below:

* [Vuetify Contributing Guide](https://vuetifyjs.com/getting-started/contributing/)
* [Contributing Rules](https://github.com/vuetifyjs/vuetify/blob/master/.github/CONTRIBUTING.md)
* [Codepen](https://template.vuetifyjs.com)
* [Codepen w/Router](https://codepen.io/zikeji/pen/ypeQNm)
* A **simple** and **small** repository

Keep in mind we cannot debug a personal application. Thank you for your understanding. If you have any additional questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).

**Comment by olivierkrener** [comment]
Ok, I may not have been clear in my explanation. Let me try again.

**A v-combobox allows filtering and selecting items from a list. I would expect that it would render each item only once and then it would show or hide them based on the filter. This is not what I am seeing, hence this issue.**

The example Codepen shows a simple v-combobox with a custom "item" slot and "selection" slots. I added several counters just to understand what is going on:
1. #mounted (on top): increments every time an item or selection slot component enters the mounted state
2. #watch (just below): increments every time a watch is triggered in these components, the watch is on a constant passed to it.
3. Then I have the same counters per item, just to have more details.

When you open the page, you see the #mounted=5 (=number of items) which is normal as all items have just been rendered. **But if you start typing and deleting you will see it keeps increasing - items are re-rendered.**. So either I am using v-combobox incorrectly or there are some extra rendering.

**Comment by KaelWD** [comment]
This is very standard behaviour for a filtered list. If you're doing something expensive in the item slot you should probably think about caching the result of it somewhere to avoid this.
